eduNEXT/xblock-mindmap
# feat: add the ability to configure the block as scorable from Studio

### Description
This PR adds the ability to configure the component of the XBlock as scorable from Studio.

### How To Test
1. Install this branch in your environment.
2. Add a block in a unit from Studio.
3. As an instructor, you should be able to select the component as scorable or no-scorable from the component settings in Studio.
4. As a student only the `save` button should be available.

### Screenshots

#### Studio
![image](https://github.com/eduNEXT/xblock-mindmap/assets/64033729/c3125ad2-16eb-4a6f-91d1-c4c94352bfa5)
#### LMS
![image](https://github.com/eduNEXT/xblock-mindmap/assets/64033729/e5a2b3cf-e3a8-45eb-b0f5-9ef6ac21bdf7)

### Limitations
The current implementation was based on [H5P XBlock](https://github.com/edly-io/h5pxblock), however, it has a limitation with the following use case:

1. As an instructor I define a component as `scorable=True`
2. As a student I make my submission.
3. As an instructor I grade the student's submission.
5. As a student I can see my progress in the course.
![image](https://github.com/eduNEXT/xblock-mindmap/assets/64033729/36607272-a942-4b47-b064-38bff49180e8)
6. As an instructor I define the same component as `scorable=False`
7. As a student, I no longer get the component score, but it still counts in the course progress.
![image](https://github.com/eduNEXT/xblock-mindmap/assets/64033729/60833ce1-feff-497c-b637-ede7558653d7)
